### ISSUE TITLE
Recognise tags pinned by their object hash instead of their name as secure

### ISSUE THREAD

**Issue by diabonas** [kpcyrd/archlinux-inputs-fsck#1]
archlinux-inputs-fsck currently only recognises Git sources pinned by a commit hash as secure. However as outlined in [a recent arch-dev-public mailing list post](https://lists.archlinux.org/archives/list/arch-dev-public@lists.archlinux.org/message/N2YKYZG6ZX7QZSQZLFFEBXZQMD2MVIDG/), pinning the SHA-1 hash of the tag object is as secure as pinning the commit ID:
```
_tag=1234567890123456789012345678901234567890 # git rev-parse "$tag_name"
source=("git+$url?signed#tag=$_tag")
```
Therefore [`is_commit_securely_pinned()`](https://github.com/kpcyrd/archlinux-inputs-fsck/blob/b54ba3d656f56c743ca5eb2c875c838cc8992c77/src/fsck.rs#L99) should distinguish the case when the tag "name" is a 40-digit hex ID and return true in this case.

**Comment by kpcyrd** [comment]
Fixed in ada3c9ef3af1842e6a6014d36ebe1a6170cc9e52, thanks!

**Comment by diabonas** [comment]
Awesome, thank you very much for the quick fix!
